### PR TITLE
helm & metrics: fix hive config properties of type stringslice

### PR DESCRIPTION
**Pull request by mhofstetter**
Hive config flags of type `StringSlice` are no longer split by space. Instead, they are split by comma. This leads to issues for config values that have been lately migrated to a Hive cell and didn't change this in the Helm files.

Therefore, this commit changes the affected config values to pass them as a comma-separated string.

* `prometheus.metrics` -> `metrics`
* `prometheus.controllerGroupMetrics` -> `controller-group-metrics`

Fixes: https://github.com/cilium/cilium/pull/26850, https://github.com/cilium/cilium/issues/28771

Thanks @giorio94 for catching this!

### CODE REVIEW

**Review by giorio94 — COMMENTED** [review]
Thanks @mhofstetter for addressing this! Overall the fix looks good to me, just two minor things:
* IIRC there's also a mention of "space separated value" in the values file about the metrics, which could also use an update.
* The metrics fix should be split in a separate PR, as it needs to be backported to v1.14

**Comment by mhofstetter** [comment]
> * IIRC there's also a mention of "space separated value" in the values file about the metrics, which could also use an update.

Found it -> changed :pray: 

> * The metrics fix should be split in a separate PR, as it needs to be backported to v1.14

Good idea - I opened a separate PR for the ingress (&L7-LB) related properties: https://github.com/cilium/cilium/pull/28922

**Review by sayboras — APPROVED** [review]
LGTM for helm changes :heavy_check_mark:

**Review by learnitall — APPROVED** [review]
LGTM for docs!

**Review by giorio94 — APPROVED** [review]
Thanks! Just a note that controller group metrics weren't present in v1.14 AFAIR, which might cause conflicts if we backport this (although they would be easy to resolve).

**Comment by joamaki** [comment]
Let's wait a bit on the decision on whether or not we want to change helm values or whether we want to `cell.Config` parsing to match what we did earlier with config map `StringSlice`.

File: install/kubernetes/cilium/values.yaml.tmpl [hunk 1]
```diff
@@ -1926,8 +1926,7 @@ prometheus:
     trustCRDsExist: false
 
   # -- Metrics that should be enabled or disabled from the default metric list.
-  # The list is expected to be separated by a space. (+metric_foo to enable
-  # metric_foo , -metric_bar to disable metric_bar).
+  # (+metric_foo to enable metric_foo , -metric_bar to disable metric_bar).
```
Comment (asauber) [inline install/kubernetes/cilium/values.yaml.tmpl]: It should be noted here that the value is an array, with the example matching that format.

```
    metrics:
      - +metric_foo
      - -metric_bar
  
This enables `metric_foo` and disables `metric_bar`.
```

**Review by joamaki — APPROVED** [review]

**Review by asauber — CHANGES_REQUESTED** [review]
This documentation update should be made before merging

https://github.com/cilium/cilium/pull/28919#discussion_r1383628055

File: install/kubernetes/cilium/values.yaml.tmpl (hunk 1 above)
Comment (mhofstetter) [reply install/kubernetes/cilium/values.yaml.tmpl]: Updated. Changed the type of the property `metrics` from `string` to `array` by setting the default from `~` to `[]`.

This shouldn't change any behaviour. It's mainly that it's displayed properly in the docs too.

**Comment by mhofstetter** [comment]
Added upgrade guide nodes and addressed input from @asauber .

In addition i changed the type of the property `metrics` from `string` to `array` by setting the default from `~` to `[]`.

This shouldn't change any behaviour. It's mainly that it's displayed properly in the docs too. The actual type was always an array of metrics.

**Review by asauber — APPROVED** [review]

**Comment by joestringer** [comment]
I am concerned about this approach breaking users in a very subtle way simply because we are refactoring internals and haven't spent time trying to minimize user pain. Related: https://github.com/cilium/cilium/issues/29210

**Review by joestringer — CHANGES_REQUESTED** [review]
We should solve this comprehensively ref. https://github.com/cilium/cilium/issues/29210 .

**Pull request closed without merging**